File: backend/backend/ApiClients/RapidApi/RapidApiApiResponseNews.cs
using System.Text.Json.Serialization;

namespace backend.ApiClients.RapidApi;

public class RapidApiApiResponseNews
{
    [JsonPropertyName("title")]
    public string Title { set; get; }
    [JsonPropertyName("publisher")]
    public string Publisher { set; get; }
    [JsonPropertyName("link")]
    public string Link { set; get; }
    [JsonPropertyName("providerPublishTime")]
    public long PublishTime { set; get; }
    [JsonPropertyName("thumbnail")]
    public RapidApiApiResponseThumbnail Thumbnail { set; get; }
    
    public string GetThumbnail() {
        return Thumbnail.Resolutions[0].Url;
    }
}

File: backend/src/main/java/io/kongkham/kcurr/RapidApiApiResponseNews.java
package io.kongkham.kcurr;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.annotation.Nullable;
import lombok.Data;

@Data
public class RapidApiApiResponseNews {
    private String title;
    private String publisher;
    private String link;
    @JsonProperty("providerPublishTime")
    private long publishTime;
    @Nullable
    private RapidApiApiResponseThumbnail thumbnail;

    public String getThumbnail() {
        return thumbnail.getResolutions()[0].getUrl();
    }
}
